### DISCUSSION TITLE
Server to server requests

### DISCUSSION THREAD

**Discussion by corygibbons** [Shopify/storefront-api-feedback#89] [Ideas & Requests]
Is there any more information about the plans to deal with server→server requests and not exhaust the ip based rate limit? Sounds like with the release of Hydrogen this will happen, from shopify/hydrogen#240:

> We recommend calling the Storefront API on the server. Hydrogen will soon introduce a mechanism to make server-to-server calls without exhausting rate limits, so it's best to structure your apps this way from the start.

However the big question I have is can we count on this to work outside of Hydrogen/Oxygen? eg. using the API directly on a Next/Vercel or Remix/CF Workers storefront?

Being able to use the Storefront API server side would provide so much more flexibility when creating custom e-commerce storefronts. 🤞

**Comment by johncraigcole** [comment]
Hey @corygibbons 👋  Yes, Shopify is looking to support server to server requests outside of a Hydrogen & Oxygen context. Nothing more than that to share today. 😄 As more details emerge we'll be sure to share them with the community.

**Reply by corygibbons** [reply]
Excellent, happy to hear that! Thanks for the response and I look forward to future updates 😃

**Reply by limorl** [reply]
Good news! We have similar issue. We are building an Alexa skill for a customer stores, and we use the storefront API.
The problem is that all calls are executed from our server, and we hit rate limits right away.
